sites/all/modules/reol_prevent_unpublish/reol_prevent_unpublish.module

```diff
@@ -44,10 +43,27 @@ function reol_prevent_unpublish_node_presave($node) {
  */
 function reol_prevent_unpublish_form_alter(&$form, &$form_state, $form_id) {
   if (strpos($form_id, '_node_form') !== FALSE) {
-    $nids = array_map('trim', explode(',', variable_get('reol_prevent_unpublish_nid', '')));
-    if (isset($form['nid']['#value']) && in_array($form['nid']['#value'], $nids)) {
+    if (isset($form['nid']['#value']) && _reol_prevent_unpublish_is_protected($form['nid']['#value'])) {
       $form['options']['status']['#default_value'] = 1;
-      $form['options']['status']['#disabled'] = true;
+      $form['options']['status']['#disabled'] = TRUE;
     }
   }
 }
+
+/**
+ * Check if the node is protected.
+ *
+ * @param $nid
+ *   The node id to check.
+ *
+ * @return bool
+ *   TRUE if protected else FALSE.
+ */
+function _reol_prevent_unpublish_is_protected($nid) {
+  $nids = array_map('trim', explode(',', variable_get('reol_prevent_unpublish_nid', '')));
+  if (in_array($nid, $nids)) {
+    return TRUE;
+  }
+
+  return FALSE;
```

Comment:
This should be written as `return in_array($nid, $nids);` (cf. https://ucsb-cs16.github.io/topics/antipatterns/#example).

Reply:
Fixed